participants/qareports.py

```diff
@@ -12,63 +48,127 @@ def handle_wi_control(self, ctrl):
-                    result_xmls.append((result, result_contents))
-                else:
-                    attachments.append((result, result_contents))
-            else:
-                attachments.append((result, open(result).read()))
-
-        if result_xmls:
-            url, msg = self.uploader.send_files(
-                result_xmls,
-                attachments,
-                hwproduct=hwproduct,
-                testtype=testtype,
-                target=target,
-                release_version=release_version,
-                build=build
-            )
 
+        url, msg = self.upload(
+            file_list=file_list,
+            import_params=dict(
+                hwproduct=wid.fields.qa.hwproduct,
+                testtype=wid.fields.qa.testtype,
+                target=wid.fields.qa.target,
+                release_version=wid.fields.qa.release_version,
+                build_txt=wid.fields.image.image_url,
+            ),
+            report_id=wid.params.report_id,
+        )
+
+        if url:
+            if not wid.fields.qa.results:
+                wid.fields.qa.results = {}
             wid.fields.qa.results.report_url = url
-            if url and ((f.qa and f.qa.move_results) or (p.move_results)):
-                move_results_dir(
-                    f.qa.results.results_dir,
-                    os.path.basename(url)
-                )
 
+            if wid.params.move_results:
+                report_id = os.path.basename(url)
+                move_results_dir(results_dir, report_id)
+
+        if msg:
             if not wid.fields.msg:
                 wid.fields.msg = []
-
             wid.fields.msg.append(msg)
+
+    def upload(self, file_list=None, import_params=None, report_id=None):
+        params = dict(auth_token=self._auth_token)
+        if report_id:
+            selector = "update/%s" % report_id
+        else:
+            selector = "import"
+            params.update(import_params)
+
+        result_index = 1
+        attachment_index = 1
+        files = {}
+        for file_name in file_list:
+            if file_name.endswith(".xml"):
+                key = "report.%s" % result_index
+                result_index += 1
+            else:
+                key = "attachment.%s" % attachment_index
+                attachment_index += 1
+            files[key] = [
+                file_name, open(file_name, 'rb'), get_content_type(file_name)
+            ]
+
+        self.log.info(
+            "Uploading results to QA-reports tool: %s", self._apiurl
+        )
+
+        response = requests.post(
+            '%s/%s' % (self._apiurl, selector),
+            files=files,
+            data=params,
+            auth=self._auth,
+        )
+        response.raise_for_status()
+        result = response.json()
+        if result.get("ok") == "1":
+            url = result.get("url", "")
+            msg = "Results uploaded successfully %s" % url
```

Comment:
if url turns out to be empty this msg won't look like there's a blank or missing 'url'.
Maybe "Results uploaded successfully to url: %s" % url   ??